R/simulate_design.R

```diff
@@ -69,7 +69,7 @@ simulate_design <- function(..., sims = 500) {
     # if you provide a list of sims for each design, i.e.
     #   sims = list(my_design_1 = c(100, 1, 1), my_design_2 = 200)
     # use it! otherwise, create a list of length designs that repeats the sims
-    if (!is_list(sims)) {
+    if (!inherits(sims, "list")) {
```

Comment:
be careful here, seems potentially edgy

Reply:
This scares me also; why not just do the diagnosis separately...

 # if you provide a list of sims for each design, i.e.
 #   sims = list(my_design_1 = c(100, 1, 1), my_design_2 = 200)
# use it! otherwise, create a list of length designs that repeats the sims

Reply:
Regarding tests. The kind of problem we have if we takes simulations from other levels arises in situations like the below. 

We get very different answers every time but with a lot of misplaced confidence in the answers because the draws below the top level are not independent. 

In  a case like the below if we used the top level we should really warn people that the bootstrapping is based on draws from a single simulation!

I suggest a warning of that form f the top level is <100, say

```
library(DeclareDesign)
U <- declare_population(
  world = add_level(N = 1, tau = sample(0:1,1)),
  samp  = add_level(N = 10, Y_Z_0 = rnorm(N)/100, Y_Z_1 = 1)
)

Z <- declare_assignment()

R <- declare_reveal()

Q <- declare_estimand(inq = mean(tau))

A <- declare_estimator(Y~Z, estimand = "inq")


des <- U + Z + R + Q + A 

set.seed(1)
diagnose_design(des,  sims = c(1, 100, 1, 1, 1))

set.seed(2)
diagnose_design(des,  sims = c(1, 100, 1, 1, 1))
```